### ISSUE TITLE
Better customizability of thank-you page

### ISSUE THREAD

**Issue by fulv** [jazkarta/jazkarta.shop#56]
I'd like to propose a modest improvement to the thank you page that would make it more customizable.  

Basically, my proposal consists of:

- a marker interface `IThankYouView`
- a browser page that inherits from `CheckoutFormBase` and that implements `IThankYouView` and otherwise does nothing except to use the same template `checkout_thankyou.pt` that is already being rendered.
- replacing the calls to `self.thankyou_template` with a call to the new view.

Basically, it would look similar to the `IStripeEnabledView`.

This would allow me to register a viewlet only for the thank you page, which I need in order to add an analytics-type script to it.

I have considered using the `after_checkout_callback_url`, but I don't really want a redirect, because that would make me lose the current state and I want the thank you page to show the same information it does now.

**Comment by skleinfeldt** [comment]
@fulv thanks for the suggested contribution!

@davisagli do you have any input on this? I feel like you're the main architect even though you're not so involved now. Thanks!

**Comment by davisagli** [comment]
The overall idea sounds reasonable, but keep in mind that the template needs access to the "old_cart" which is currently set on the view when the form submission is processed, so you'll need to find another way to pass that to the template.

**Comment by fulv** [comment]
Yeah, I realized that when I tried out my implementation.
